sdk/identity/Azure.Identity/src/DefaultAzureCredentialFactory.cs

```diff
@@ -32,70 +29,104 @@ protected DefaultAzureCredentialFactory(DefaultAzureCredentialOptions options, C
 
         public TokenCredential[] CreateCredentialChain()
         {
-            if (_useDefaultCredentialChain)
-            {
-                return s_defaultCredentialChain;
-            }
-
-            List<TokenCredential> chain = new(10);
+            string credentialSelection = EnvironmentVariables.CredentialSelection?.Trim();
+            bool _useDevCredentials = Constants.DevCredentials.Equals(credentialSelection, StringComparison.OrdinalIgnoreCase);
+            bool _useProdCredentials = Constants.ProdCredentials.Equals(credentialSelection, StringComparison.OrdinalIgnoreCase);
 
-            if (!Options.ExcludeEnvironmentCredential)
+            if (credentialSelection != null && !_useDevCredentials && !_useProdCredentials)
             {
-                chain.Add(CreateEnvironmentCredential());
+                throw new InvalidOperationException($"Invalid value for environment variable AZURE_TOKEN_CREDENTIALS: {credentialSelection}. Valid values are 'dev' or 'prod'.");
```

Comment:
Should it print the value from the `ProdCredentials` and `DevCredentials` consts instead of hardcoding it?